samples/GStreamerWPF/GStreamerControl.Library/Playback.cs

```diff
@@ -253,96 +266,96 @@ protected bool CreateElements()
+					_avdec["max-threads"] = 12;
+				}
 			}
-			catch (Exception ex)
-			{
+			catch (Exception ex) {
 				Log(ex.ToString(), LogLevelFlags.Critical, ex);
 				return false;
 			}
 
 			return true;
 		}
 
-		private void OnBusMessage(object o, MessageArgs margs)
-        {
-            try
-            {
+		private void OnBusMessage(object o, MessageArgs margs) {
+			try {
 
-                Message message = margs.Message;
-
-                switch (message.Type)
-                {
-                    case MessageType.Eos:
-
-                        Log("Replaying stream...", LogLevelFlags.Info);
-
-                        var ret = _pipeline.SetState(Gst.State.Ready);
-
-                        if (ret == StateChangeReturn.Async)
-                            ret = _pipeline.GetState(out var state, out var pending, Gst.Constants.SECOND * 10L);
-
-                        if (ret == StateChangeReturn.Success)
-                        {
-                            ret = _pipeline.SetState(Gst.State.Playing);
-                            if (ret == StateChangeReturn.Async)
-                                ret = _pipeline.GetState(out var state, out var pending, Gst.Constants.SECOND * 10L);
-                        }
-                        break;
-                    case MessageType.Error:
-                        message.ParseError(out GException err, out string debug);
-                        if (debug == null)
-                        {
-                            debug = "none";
-                        }
-                        Log($"Error! Bus message: {debug}", LogLevelFlags.Error, err);
-                        if (err.Code == 3)
-                        {
-                            System.Windows.Application.Current.Dispatcher.InvokeShutdown();
-                        }
-                        break;
-                }
+				Message message = margs.Message;
 
-            }
-            catch (Exception ex)
-            {
-                Log("Bus message error.", LogLevelFlags.Error, ex);
-            }
-        }
+				switch (message.Type) {
+					case MessageType.Eos:
 
-		private void VideoSink_OnBeginDraw(object o, GLib.SignalArgs args)
-		{
+						Log("Replaying stream...", LogLevelFlags.Info);
+
+						var ret = _pipeline.SetState(Gst.State.Ready);
+
+						if (ret == StateChangeReturn.Async)
+							ret = _pipeline.GetState(out var state, out var pending, Gst.Constants.SECOND * 10L);
+
+						if (ret == StateChangeReturn.Success) {
+							ret = _pipeline.SetState(Gst.State.Playing);
+							if (ret == StateChangeReturn.Async)
+								ret = _pipeline.GetState(out var state, out var pending, Gst.Constants.SECOND * 10L);
+						}
+						break;
+					case MessageType.Error:
+						message.ParseError(out GException err, out string debug);
+						if (debug == null) {
+							debug = "none";
+						}
+						Log($"Error! Bus message: {debug}", LogLevelFlags.Error, err);
+						if (err.Code == 3) {
```

Comment:
I realize this isn't your code, but what is this? Perhaps we should just drop it?